### DISCUSSION TITLE
Accessing a widget from a javascript

### DISCUSSION THREAD

**Discussion by mbenson1** [yamcs/yamcs-studio#100] [Q&A] [answered]
How can I read the value of a widget from within a javascript?  I have opi's and javascripts from an old release of Yamcs Studio (4 or 5 years old).  The following script used to work:

importPackage(Packages.org.csstudio.opibuilder.scriptUtil);
importPackage(Packages.org.yamcs.studio.script);
importPackage(Packages.org.csstudio.simplepv);

var strMode = VTypeHelper.getString(display.getWidget('inMode').getPropertyValue('pv_value'));
var cmd = '/cfs/cfe_es/SetSysLogMode(Mode: "' + strMode + '")';
Yamcs.issueCommand(cmd);

But now the "var strMode" line errors out indicating "VTypeHelper" is undefined.  What's the right way to do this?

**Comment by fqqb** [comment, marked as answer]
Can you try:
`importPackage(Packages.org.yamcs.studio.data);`

that’s where VTypeHelper is located these days

For issueCommand we also have a more convenient variant (subject to preference), to do it like this:

```
Yamcs.issueCommand('/cfs/cfe_es/SetSysLogMode', {
  'Mode': strMode
});
```

i.e. no string concats

**Reply by mbenson1** [reply]
Those two changes worked perfectly.  Thanks.
